numpy/core/src/umath/loops.c.src

```diff
@@ -2127,15 +2114,13 @@ HALF_nextafter(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNU
  * #kind = maximum, minimum#
  * #OP =  npy_half_ge, npy_half_le#
  **/
-NPY_NO_EXPORT void
+NPY_NO_EXPORT NPY_GCC_OPT_3 void
 HALF_@kind@(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED(func))
 {
     /*  */
-    BINARY_LOOP {
-        const npy_half in1 = *(npy_half *)ip1;
-        const npy_half in2 = *(npy_half *)ip2;
-        *((npy_half *)op1) = (@OP@(in1, in2) || npy_half_isnan(in1)) ? in1 : in2;
-    }
+    BINARY_LOOP_FAST(npy_half, npy_half,
```

Comment:
similar to datetime, performance of float is probably not a big concern it is a input/output format, computations should be done on casted data to float32

Reply:
Thanks, removed